python/pyarrow/pandas_compat.py

```diff
@@ -1080,8 +1083,10 @@ def _reconstruct_columns_from_metadata(columns, column_indexes):
     ]
 
     # Convert each level to the dtype provided in the metadata
+    # ARROW-9096: need numpy_type to match cast against original DataFrame
```

Comment:
nit: drop JIRA reference.  maybe also the whole comment since this is used directly below?

Reply:
makes sense - it's easy to see what's going on w/o this line so dropped it as suggested